File: Solutions/_11.cs
public class Solution
{
    public int MaxArea(int[] height)
    {
        int left = 0;
        int right = height.Count() - 1;
        int maxWater = 0;
        while (left < right)
        {
            int leftValue = height[left];
            int rightValue = height[right];
            int currentWater = Math.Min(height[left], height[right]) * (right - left);
            maxWater = Math.Max(maxWater, currentWater);
            if (height[left] > height[right])
            {
                right--;
            }
            else
            {
                left++;
            }
        }
        return maxWater;
    }
}

File: Solutions/_11.go
func maxArea(height []int) (maxWater int) {
	left := 0
	right := len(height) - 1

	for left < right {
		currentWater := min(height[left], height[right]) * (right - left)
		maxWater = max(maxWater, currentWater)
		if height[left] > height[right] {
			right--
		} else {
			left++
		}
	}
	return
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
